mempool/types/src/tx_status.rs

```diff
@@ -21,16 +21,45 @@
 pub enum TxStatus {
     /// Transaction is in mempool
     InMempool,
+
+    /// Transaction has already been in the mempool, duplicate insertion
+    InMempoolDuplicate,
+
     /// Transaction is in orphan pool
     InOrphanPool,
+
+    /// Transaction has already been in the orphan pool, duplicate insertion
+    InOrphanPoolDuplicate,
```

Comment:
Just to be clear, this is much better than before, but this may warrant changes in p2p and the wallet to expect different behavior. I'll notify the guys about this.

Reply:
I made some changes to p2p, I think it's fine now. Not 100% sure about the wallet.

Reply:
I don't see any problems in p2p either.

Reply:
All good. I just want to make sure all related parties agree about the change in the interface behavior that only affects runtime without having unexpected side-effects. That's the scariest thing in software. Cheers!